packages/vue/src/generators/application/lib/normalize-options.ts

```diff
@@ -0,0 +1,58 @@
+import { Tree, extractLayoutDirectory, names } from '@nx/devkit';
+import { determineProjectNameAndRootOptions } from '@nx/devkit/src/generators/project-name-and-root-utils';
+import { NormalizedSchema, Schema } from '../schema';
+import { findFreePort } from '@nx/js';
+
+export function normalizeDirectory(options: Schema) {
+  options.directory = options.directory?.replace(/\\{1,2}/g, '/');
+  const { projectDirectory } = extractLayoutDirectory(options.directory);
+  return projectDirectory
+    ? `${names(projectDirectory).fileName}/${names(options.name).fileName}`
+    : names(options.name).fileName;
+}
+
+export async function normalizeOptions(
+  host: Tree,
+  options: Schema,
+  callingGenerator = '@nx/vue:application'
+): Promise<NormalizedSchema> {
+  const {
+    projectName: appProjectName,
+    projectRoot: appProjectRoot,
+    projectNameAndRootFormat,
+  } = await determineProjectNameAndRootOptions(host, {
+    name: options.name,
+    projectType: 'application',
+    directory: options.directory,
+    projectNameAndRootFormat: options.projectNameAndRootFormat,
+    rootProject: options.rootProject,
+    callingGenerator,
+  });
+  options.rootProject = appProjectRoot === '.';
+  options.projectNameAndRootFormat = projectNameAndRootFormat;
+
+  const e2eProjectName = options.rootProject ? 'e2e' : `${appProjectName}-e2e`;
+  const e2eProjectRoot = options.rootProject ? 'e2e' : `${appProjectRoot}-e2e`;
+
+  const parsedTags = options.tags
+    ? options.tags.split(',').map((s) => s.trim())
+    : [];
+
+  const normalized = {
+    ...options,
+    name: names(options.name).fileName,
+    projectName: appProjectName,
+    appProjectRoot,
+    e2eProjectName,
+    e2eProjectRoot,
+    parsedTags,
+  } as NormalizedSchema;
+
+  normalized.style = options.style ?? 'css';
+  normalized.routing = normalized.routing ?? false;
+  normalized.unitTestRunner ??= 'vitest';
+  normalized.e2eTestRunner = normalized.e2eTestRunner ?? 'cypress';
+  normalized.devServerPort ??= findFreePort(host);
```

Comment:
Do we need to use `findFreePort` here? We normally don't expect for people to run multiple `apps` at once UNLESS it is a MFE `host/remote`.

Reply:
Sure, we can remove it. The original thinking was to make it at parity with React.

Reply:
Sure, we can remove it. The original thinking was to make it at parity with React and Angular.